draft-ietf-tls-esni.md

```diff
@@ -438,10 +426,24 @@ echo_hrr_key = context.Export("tls13-echo-hrr-key", 16)
 ~~~
 
 Note that the HPKE algorithm identifiers are those which match the client's
-chosen Ciphersuite, according to {{hpke-map}}. The client MAY replace any large,
+chosen CipherSuite, according to {{hpke-map}}. The client MAY replace any large,
 duplicated extensions in ClientHelloInner with the corresponding "outer_extensions"
 extension, as described in {{outer-extensions}}.
 
+The client then generates a ClientHelloInner value. In addition to the normal
+values, ClientHelloInner MUST also contain:
+
+ - an "echo_nonce" extension
+ - TLS padding {{!RFC7685}}
+
+The padding SHOULD contain X bytes of padding, where X = ECHOConfig.maximum_name_length - length(dns_name),
+rounded up to the nearest multiple of 16, and dns_name is the DNS name in the
```

Comment:
The rounding should apply to the target ClientHelloInner length, not the amount of padding, right? If X is always a multiple of 16, we'll leak the unpadded size of the ClientHelloInner mod 16.

Or we could go back to asking for maximum_name_length to be rounded up, which is certainly easier to say. I dunno how much to care about the other sources of padding below and not stacking up the multiple of 16 roundings? (But I guess the point of the multiple of 16 is to try to reduce the overall distribution of lengths, in which case you do want it applied at the end?)

Reply:
> The rounding should apply to the target ClientHelloInner length, not the amount of padding, right?

That was the intent, yep, though it's not correct here! Let me fix that quickly. (We want to round X + len(dns_name) to the nearest multiple of 16.)